koku/sources/storage.py

```diff
@@ -150,6 +151,17 @@ def load_providers_to_delete():
     return providers_to_delete
 
 
+def get_source(source_id, err_msg):
```

Comment:
Question with this: If the DB is down and we hit the close, I know it will reopen a new connection the next time it needs to, but what about the current call will we potentially drop info? Would it be worth thinking about a retry in this get source method that tries a few times to get the Sources.object.get... if we do hit the InterfaceError?

Reply:
I was thinking about that. I could add some retry logic.

Though, I need to look further into what happens to an unsuccessful message. There are a few flags here and there. Need to check on what they do and how they work. I think with what I have right now, the message would be lost, which would not be good.

Reply:
I don't mind doing this incrementally, if we want to merge this PR and then look at retry logic next.